audio_service/lib/audio_service.dart

```diff
@@ -914,40 +916,76 @@ class AudioService {
     return handler;
   }
 
+  static CancelableOperation<Object?>? _artFetchOperation;
+  static late double _debugCurrentFetchId;
   static Future<void> _observeMediaItem() async {
     await for (var mediaItem in _handler.mediaItem) {
       if (mediaItem == null) continue;
+      _artFetchOperation?.cancel();
       final artUri = mediaItem.artUri;
       if (artUri != null) {
-        // We potentially need to fetch the art.
-        String? filePath;
+        /// Sends media item to the platform.
+        /// We potentially need to fetch the art before that.
+        Future<void> _sendToPlatform(String? filePath) async {
+          final extras =
+              Map<String, dynamic>.of(mediaItem.extras ?? <String, dynamic>{});
+          extras['artCacheFile'] = filePath;
+          final platformMediaItem = mediaItem.copyWith(extras: extras);
```

Comment:
This might be a little cleaner:

```dart
final platformMediaItem = mediaItem.copyWith(extras: {
  if (mediaItem.extras != null) ...mediaItem.extras!,
  'artCacheFile': filePath,
});
```

Implicit dynamic might bite here, I'm also thinking of later switching to flutter's new lints.

Reply:
Yes it seems better (commenting from mobile so can't push it)